File: 217-contains-duplicate/contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
const containsDuplicate = (nums) => {
    let hashSet = new Set()
    for(i = 0; i < nums.length; i++) {
        if (hashSet.has(nums[i])) {
            return true;
        }
        hashSet.add(nums[i]);
    }
       
      return false
};

File: 217-contains-duplicate/contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # inputs: array of nums, 
        # return true if any value appears twice
        # return false if every element is distinct

        # i can create a hash and if the element appears more than once i can return true
        myHash = {}
    
        for i in range(len(nums)):
            if nums[i] in myHash:
                return True
            myHash[nums[i]] = 1
        return False


        